examples/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -27,68 +32,130 @@
     std::chrono::high_resolution_clock::time_point _tick;
 
   public:
-    Timer(entry_reference_type ref)
-        : _entry{ref}
-        , _started{false}
+    Timer(const MPI_Comm mpi_communicator = MPI_COMM_WORLD)
+        : _mpi_comm(mpi_communicator)
     {
     }
+
+    // Prevent accidental copy construction when calling
+    // TimeMonitor::getTimer(). The obtained timer would not be registered in
+    // PerformanceMonitor.
+    Timer(Timer const &) = delete;
+
+    // Reset the lap timer to start measuring a new iteration.
     void start()
     {
       assert(!_started);
       _tick = std::chrono::high_resolution_clock::now();
       _started = true;
     }
+
+    // Take the maximum of the elapsed lap time across all the MPI processes and
+    // store it in the boost::accumulator object.
     void stop()
     {
       assert(_started);
       std::chrono::duration<double> duration =
           std::chrono::high_resolution_clock::now() - _tick;
-      // NOTE I have put much thought into whether we should use the
-      // operator+= and keep track of how many times the timer was
-      // restarted.  To be honest I have not even looked was the original
-      // TimeMonitor behavior is :)
-      _entry.second = duration.count();
+      MPI_Allreduce(MPI_IN_PLACE, &duration, 1, MPI_DOUBLE, MPI_MAX, _mpi_comm);
+      _statistics(duration.count());
       _started = false;
     }
+
+    // Provide read access to the stored boost::accumulator_set object.
+    data_type const &get_statistics() const { return _statistics; }
+
+    // Following CppCon 2015: Bryce Adelstein-Lelbach “Benchmarking C++ Code"
+    // (https://www.youtube.com/watch?v=zWxSZcpeS8Q&t=1780
+    // https://github.com/CppCon/CppCon2015/tree/master/Presentations/Benchmarking%20C%2B%2B%20Code
+    // (slide 37)) given a relative margin of error e_m, a quantile z
+    // corresponding to the given confidence to be achieved, the (estimated)
+    // mean \nu and the (estimated) standard deviation \sigma, the number of
+    // required samples n can be computed as n = ((z \sigma)/(e_m/2 \mu))^2.
+    //
+    // Return this estimate.
+    int estimate_required_sample_size(double confidence,
+                                      double relative_error_margin) const
+    {
+      int const n_measurements = ba::count(_statistics);
+      boost::math::students_t const dist(n_measurements - 1);
+      double const z =
+          boost::math::quantile(complement(dist, (1 - confidence) / 2));
+
+      // We need to scale the sample standard deviation to get an unbiased
+      // estimator of the population standard deviation.
+      double const current_stddev = std::sqrt(
+          ba::variance(_statistics) * (n_measurements / (n_measurements - 1.)));
+      double const current_mean = ba::mean(_statistics);
+      auto const tmp =
+          z * current_stddev / (relative_error_margin * current_mean / 2.);
+      return static_cast<int>(std::ceil(tmp * tmp));
```

Comment:
So far I couldn't find parameters for `boost::math::students_t::find_degrees_of_freedom` such that the returned value would match the one computed here.
```
boost::math::students_t::find_degrees_of_freedom(relative_error_margin*current_mean, confidence/2, .9999999, current_stddev)
```
is just 1.3 times larger though.